app/models/user.rb

```diff
@@ -9,11 +9,10 @@ class User < ApplicationRecord
   FAVORITE_AUTHORS_INCLUDES = [:books, :favorites,
 {image_attachment: :blob}].freeze
 
-  has_secure_password
-  # has_secure_password cung cấp: # rubocop:disable Style/AsciiComments
-  # - Các thuộc tính ảo: password, password_confirmation # rubocop:disable Style/AsciiComments
-  # - Trường password_digest để lưu hash # rubocop:disable Style/AsciiComments
-  # - Phương thức authenticate(password) để xác thực # rubocop:disable Style/AsciiComments
+  devise :database_authenticatable, :rememberable, :validatable,
+         :confirmable, :lockable, :recoverable, :omniauthable,
+         :registerable
```

Comment:
```suggestion
  devise :database_authenticatable, :rememberable, :validatable,
         :lockable, :recoverable, :registerable
```
Dùng các tính năng tương ứng, tránh thừa code